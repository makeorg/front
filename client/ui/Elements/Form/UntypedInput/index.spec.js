import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { UntypedInput } from './index';

jest.mock('Client/ui/Elements/Form/Styled/Input', () => ({
  BasicInputStyle: 'BasicInputStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Content', () => ({
  MiddleFakeFieldStyle: 'MiddleFakeFieldStyle',
  FloatingLabelStyle: 'FloatingLabelStyle',
  FieldWrapperStyle: 'FieldWrapperStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Icons', () => ({
  CenterInputIconStyle: 'CenterInputIconStyle',
}));

const type = 'foo';
const name = 'bar';
const icon = 'SvgEnvelope';
const value = 'baz';
const label = 'qux';
const handleChange = () => {};

describe('UntypedInput', () => {
  it('must return the diff between snapshot when input is required or optionnal', () => {
    const RequiredInput = renderer
      .create(
        <UntypedInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          required
        />
      )
      .toJSON();
    const OptionnalInput = renderer
      .create(
        <UntypedInput
          type={type}
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
        />
      )
      .toJSON();
    expect(snapshotDiff(RequiredInput, OptionnalInput)).toMatchSnapshot();
  });
});
