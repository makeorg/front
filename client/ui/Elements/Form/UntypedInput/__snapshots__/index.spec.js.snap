// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UntypedInput must return the diff between snapshot when input is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -8,15 +8,15 @@
    </CenterInputIconStyle>
    <div
      className="sc-1ln9y8s-4"
    >
      <BasicInputStyle
        id="bar"
        name="bar"
        onChange={[Function handleChange]}
-       required={true}
+       required={false}
        type="foo"
        value="baz"
      />
      <label
        className="sc-1ln9y8s-9"
`;
